File: templates/service-agent/src/routes/agents.ts
/**
 * Agent Management Routes
 * 
 * Endpoints for listing and managing agents.
 */

import { FastifyPluginAsync } from 'fastify';
import { NotFoundError } from '@framework/core/errors';
import { listAgents, getAgent } from '../agents';

export const agentRoutes: FastifyPluginAsync = async (fastify) => {
  // List all agents
  fastify.get('/', async () => {
    const agents = listAgents(fastify.agents);
    
    return {
      success: true,
      data: {
        agents,
        count: agents.length,
      },
    };
  });
  
  // Get agent details
  fastify.get('/:agentId', async (request) => {
    const { agentId } = request.params as { agentId: string };
    const agent = getAgent(fastify.agents, agentId);
    
    if (!agent) {
      throw new NotFoundError('Agent', agentId);
    }
    
    return {
      success: true,
      data: {
        id: agent.id,
        name: agent.name,
        description: agent.description,
        tools: agent.tools.map(t => ({
          name: t.name,
          description: t.description,
        })),
      },
    };
  });
  
  // Get agent's available tools
  fastify.get('/:agentId/tools', async (request) => {
    const { agentId } = request.params as { agentId: string };
    const agent = getAgent(fastify.agents, agentId);
    
    if (!agent) {
      throw new NotFoundError('Agent', agentId);
    }
    
    return {
      success: true,
      data: {
        tools: agent.tools.map(t => ({
          name: t.name,
          description: t.description,
          parameters: t.parameters,
        })),
      },
    };
  });
};

File: templates/service-python/src/api/agents.py
"""
Agent Routes
============
"""

from fastapi import APIRouter, HTTPException

from src.agents import DEFAULT_AGENTS, get_agent, list_agents
from src.models import AgentInfo, ApiResponse
from src.tools import list_tools

router = APIRouter()


@router.get("", response_model=ApiResponse)
async def get_agents() -> ApiResponse:
    """List all available agents."""
    return ApiResponse(
        success=True,
        data=list_agents(),
    )


@router.get("/{agent_id}", response_model=ApiResponse)
async def get_agent_info(agent_id: str) -> ApiResponse:
    """Get details about a specific agent."""
    config = DEFAULT_AGENTS.get(agent_id)
    if not config:
        raise HTTPException(
            status_code=404,
            detail=f"Agent not found: {agent_id}",
        )

    return ApiResponse(
        success=True,
        data=AgentInfo(
            id=config.id,
            name=config.name,
            description=config.description,
            tools=config.tools,
        ).model_dump(),
    )


@router.get("/{agent_id}/tools", response_model=ApiResponse)
async def get_agent_tools(agent_id: str) -> ApiResponse:
    """Get tools available to a specific agent."""
    agent = get_agent(agent_id)
    if not agent:
        raise HTTPException(
            status_code=404,
            detail=f"Agent not found: {agent_id}",
        )

    return ApiResponse(
        success=True,
        data={
            "agent_id": agent_id,
            "tools": agent.tool_definitions,
        },
    )


@router.get("/tools/all", response_model=ApiResponse)
async def get_all_tools() -> ApiResponse:
    """List all available tools."""
    return ApiResponse(
        success=True,
        data=list_tools(),
    )
